subsys/partition_manager/CMakeLists.txt

```diff
@@ -85,6 +85,10 @@ if (CONFIG_TRUSTED_EXECUTION_NONSECURE OR CONFIG_TRUSTED_EXECUTION_SECURE)
   ncs_add_partition_manager_config(pm.yml.trustzone)
 endif()
 
+if (CONFIG_BUILD_WITH_TFM AND (CONFIG_SETTINGS_FCB OR CONFIG_SETTINGS_NVS))
```

Comment:
Could you add a comment here describing why this is necessary?

Should this code be in `modules/tfm/zephyr/CMakeLists.txt` ?

Reply:
We want to enable NS_STORAGE in TF-M iff there exists a storage partition. This CMakeLists.txt is a file which controls whether the storage partition should be created, so I thought it would also be a right place to configure the region as non-secure, no?

Reply:
The storage partition can be used with or without TF-M, and the handling we do here is TF-M specific. IMO it is better to perform the TF-M specific stuff in the appropriate files so we don't pollute other files with TF-M-only logic.

If we do this inside the `modules/tfm/zephyr/CMakeLists.txt` then we know that `CONFIG_BUILD_WITH_TFM` is set. Also it is clear for anyone that is working on TF-M that this is something that needs to be done by simply reading the CMakeLists.txt associated with TF-M, instead of grepping around the whole `nrf` repo to see where we do TF-M related stuff.

Its not a blocker, but you should strongly consider moving it at least ;)

Reply:
Sure, I can move it.